tests/test_thread.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+    tests/test_thread.cpp -- call pybind11 bound methods in threads
+
+    Copyright (c) 2021 Laramie Leavitt (Google LLC) <lar@google.com>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#include <pybind11/cast.h>
+#include <pybind11/pybind11.h>
+
+#include <chrono>
+#include <thread>
+
+#include "pybind11_tests.h"
+
+namespace py = pybind11;
+
+namespace {
+
+struct IntStruct {
+    int value;
```

Comment:
Could add a destructor that modified `value` or modifies some global variable, so that we aren't relying on debugging features of the memory allocator as much.

Separately, perhaps add a note that this test should be run with asan for greater effectiveness.

Reply:
Done.